posthog/hogql/test/test_printer.py

```diff
@@ -467,6 +467,30 @@ def test_property_groups_optimized_basic_equality_comparisons(self) -> None:
         # ... unless we can distinguish ``Nullable(Nothing)`` from ``Nullable(*)`` -- this _could_ be safely optimized.
         self._test_property_group_comparison("properties.key = lower(NULL)", None)
 
+    def test_property_groups_optimized_boolean_equality_comparisons(self) -> None:
+        PropertyDefinition.objects.create(
+            team=self.team, name="is_boolean", property_type="Boolean", type=PropertyDefinition.Type.EVENT
+        )
```

Comment:
This is technically unnecessary since the comparison to a boolean constant is what triggers this behavior.

This also means that the optimized property group mode assumes the non-constant operand as a boolean in these comparisons but doesn't do any explicit type conversion (i.e. it will evaluate the same as if the property definition type was changed via the data management UI from String to Boolean.)